From: Martin K. Petersen <martin.petersen@oracle.com>
Date: Fri, 26 Feb 2010 00:20:39 -0500
Subject: [PATCH] block: Consolidate phys_segment and hw_segment limits
References: FATE#311054
Patch-Mainline: 2.6.34

Except for SCSI no device drivers distinguish between physical and
hardware segment limits.  Consolidate the two into a single segment
limit.

Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
Automatically created from "patches.suse/block-Consolidate-phys_segment-and-hw_segment-limit.patch" by xen-port-patches.py

--- sle11sp2-2011-05-31.orig/drivers/xen/blkfront/vbd.c	2011-05-31 11:18:53.000000000 +0200
+++ sle11sp2-2011-05-31/drivers/xen/blkfront/vbd.c	2011-05-31 11:19:19.000000000 +0200
@@ -318,8 +318,7 @@ xlvbd_init_blk_queue(struct gendisk *gd,
 	blk_queue_max_segment_size(rq, PAGE_SIZE);
 
 	/* Ensure a merged request will fit in a single I/O ring slot. */
-	blk_queue_max_phys_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
-	blk_queue_max_hw_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
+	blk_queue_max_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
 
 	/* Make sure buffer addresses are sector-aligned. */
 	blk_queue_dma_alignment(rq, 511);
--- sle11sp2-2011-05-31.orig/drivers/xen/blktap2/device.c	2011-05-31 11:18:58.000000000 +0200
+++ sle11sp2-2011-05-31/drivers/xen/blktap2/device.c	2011-05-31 11:19:16.000000000 +0200
@@ -995,8 +995,7 @@ blktap_device_configure(struct blktap *t
 	blk_queue_max_segment_size(rq, PAGE_SIZE);
 
 	/* Ensure a merged request will fit in a single I/O ring slot. */
-	blk_queue_max_phys_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
-	blk_queue_max_hw_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
+	blk_queue_max_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
 
 	/* Make sure buffer addresses are sector-aligned. */
 	blk_queue_dma_alignment(rq, 511);
--- sle11sp2-2011-05-31.orig/drivers/xen/blktap2-new/device.c	2011-05-31 11:19:04.000000000 +0200
+++ sle11sp2-2011-05-31/drivers/xen/blktap2-new/device.c	2011-05-31 11:19:12.000000000 +0200
@@ -297,8 +297,7 @@ blktap_device_configure(struct blktap *t
 	blk_queue_max_segment_size(rq, PAGE_SIZE);
 
 	/* Ensure a merged request will fit in a single I/O ring slot. */
-	blk_queue_max_phys_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
-	blk_queue_max_hw_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
+	blk_queue_max_segments(rq, BLKIF_MAX_SEGMENTS_PER_REQUEST);
 
 	/* Make sure buffer addresses are sector-aligned. */
 	blk_queue_dma_alignment(rq, 511);
